project-keeper/src/main/java/com/exasol/projectkeeper/validators/changesfile/DependencySectionFixer.java

```diff
@@ -35,20 +33,12 @@ public DependencySectionFixer(final List<AnalyzedSource> sources) {
     public ChangesFile fix(final ChangesFile changesFile) {
         final List<NamedDependencyChangeReport> reports = this.sources.stream().map(this::getDependencyChangesOfSource)
                 .collect(Collectors.toList());
-        final List<String> renderedReport = new DependencyChangeReportRenderer().render(reports);
-        final List<ChangesFileSection> sections = new ArrayList<>(changesFile.getSections());
-        removeDependencySection(sections);
-        if (!renderedReport.isEmpty()) {
-            sections.add(new ChangesFileSection(renderedReport));
-        }
-        return new ChangesFile(List.copyOf(changesFile.getHeaderSectionLines()), sections);
+        final Optional<ChangesFileSection> dependencyChanges = new DependencyChangeReportRenderer().render(reports);
+        return changesFile.toBuilder() //
+                .dependencyChangeSection(dependencyChanges.orElse(null)).build();
     }
 
     private NamedDependencyChangeReport getDependencyChangesOfSource(final AnalyzedSource source) {
         return new NamedDependencyChangeReport(source.getProjectName(), source.getDependencyChanges());
     }
-
-    private void removeDependencySection(final List<ChangesFileSection> sections) {
-        sections.removeIf(section -> section.getHeading().compareToIgnoreCase(DEPENDENCY_UPDATES_HEADING) == 0);
-    }
 }
```

Comment:
These simplifications seem to be the fruit of the work invested into the other classes.
Great!

Reply:
:tada: